game/src/edit/roads.rs

```diff
@@ -207,6 +209,40 @@ impl State<App> for RoadEditor {
                     assert!(self.current_lane.is_none());
                     self.current_lane = Some(app.primary.map.get_r(self.r).lanes_ltr()[idx].0);
                     self.recalc_all_panels(ctx, app);
+                } else if x == "Access restrictions" {
+                    // The RoadEditor maintains an undo/redo stack for a single road, but the
```

Comment:
In the common case of opening the road editor then clicking access restrictions, this is fine; we jump right to the zone editor. But if the user modifies a lane, then clicks this button, we ask to confirm. It's kind of unexpected; clicking access restrictions totally takes us away from the current road editor UI, destroying the undo/redo stack. I don't think a more clever implementation is possible, because the zone editor applies to multiple roads. If you confirmed changes there and came back to the road editor, what should we do?

Ideally we'd enter the zone editor another way, unrelated to this road editing UI. But I'm not sure where to squeeze it in.

Reply:
I think I see what you're saying... from a data perspective they're currently quite different activities. 

The access restrictions are likely to be applied to a long stretch of segments, and apply to the entire road, not an individual segment, so we want the user to be able to quickly "paint" the same restriction over potentially a bunch of different roads.

I think we actually want some kind of a similar flow for a user to "paint" the same lane edits along an easily selectable stretch of road, but it's just a harder problem to solve because the edits we want to copy around aren't always easily applicable if the underlying lane config doesn't match.

So anyway, after trying it out, I think the flow you've come up with is pretty clear for now. I wasn't confused as to why I was being prompted to save my lane-edits before leaving the lane-edits-mode and entering access-edits-mode. 👍

Reply:
Hmm... a maybe related crash though...

I'm not exactly the sequence of actions, but I think it was something like:
1. I'd previously made some lane edits
2. I'd launched the access control editor, I don't think I made any changes before hitting cancel
3. crash

I can't immediately reproduce with those steps though.

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', game/src/edit/roads.rs:521:52
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Reply:
Ah, `lane_type_to_icon` is missing a case for shared turn lanes. I'll fix that up!